### PR TITLE
BF 2276: pass path_src into commit as well (so we commit renamed files removal as well)

### LINKED ISSUE #2276: datalad run  misses original files which were renamed and leaves them deleted
#### What steps will reproduce the problem?
```shell
$> datalad create /tmp/testrunrename                           
[INFO   ] Creating a new annex repo at /tmp/testrunrename 
create(ok): /tmp/testrunrename (dataset)                                                                                                                                                                                                      

$> cd /tmp/testrunrename

$> echo 123 > 123; datalad add 123
add(ok): /tmp/testrunrename/123 (file)                                                                                                                                                                                                        
save(ok): /tmp/testrunrename (dataset)                                                                                                                                                                                                        
action summary:
  add (ok: 1)
  save (ok: 1)

$> datalad run "git mv 123 124"
[INFO   ] == Command start (output follows) ===== 
[INFO   ] == Command exit (modification check follows) ===== 
add(notneeded): 124 (file) [already included in the dataset]
save(ok): /tmp/testrunrename (dataset)
action summary:
  add (notneeded: 1)
  save (ok: 1)

$> git status
On branch master
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	deleted:    123

```
#### What version of DataLad are you using (run `datalad --version`)? On what operating system (consider running `datalad plugin wtf`)?
0.9.2

### PR DESCRIPTION
**Pull request by yarikoptic**
This pull request fixes #2276

This pull request proposes to provide `path_src` portions of the annotated paths while doing the commit in `save`.  Also while at it in aa0143b14982438392e98eb24880fc312d13f113  I did something along the same lines -- return path_src as well in annotated2content_by_ds  if only paths are requested.  I am not sure even if that line of logic gets triggered by any test/use-case.  Running all the tests will show. Otherwise it should be ready for review etc

### Changes
- [x] This change is complete

Please have a look @datalad/developers

### CODE REVIEW

File: datalad/interface/annotate_paths.py [hunk 1]
```diff
@@ -124,21 +125,28 @@ def annotated2content_by_ds(annotated, refds_path, path_only=False):
...
-                toappendto.append(r['path'] if path_only else r)
+                toappendto.append(r_path if path_only else r)
                 content_by_ds[parentds] = toappendto
         else:
             # files and dirs
             # common case, something with a parentds
             toappendto = content_by_ds.get(parentds, [])
-            toappendto.append(r['path'] if path_only else r)
+            if path_only:
+                r_path_src = r.get('path_src')
```
Comment (yarikoptic) [inline datalad/interface/annotate_paths.py]: so it seems that nowhere in the code path_only=True is used
```shell
$> git grep 'path_only=[^F]'
```
I would be happy to remove this option/logic if @mih agrees to such lobotomization ;-)

Comment (mih) [reply datalad/interface/annotate_paths.py]: This option was needed for compat with legacy code. If it is no longer needed, I would be happy to see it go.

Comment (yarikoptic) [reply datalad/interface/annotate_paths.py]: cool, then I will do it in this PR so we do not forget. thanks

**Comment by yarikoptic** [comment]
ok, RFed a bit, all green, merging

**Pull request merged**